main.nf

```diff
@@ -464,134 +519,205 @@ dwi_for_test_gibbs
     .into{dwi_for_eddy; dwi_for_topup; dwi_for_eddy_topup; dwi_for_test_eddy_topup}
 
 dwi_for_topup
-    .join(gradients_for_topup)
-    .join(rev_b0)
-    .join(readout_encoding_for_topup)
-    .set{dwi_gradients_rev_b0_for_topup}
+    .join(gradients_for_prepare_topup)
```

Comment:
The Prepare_for_Topup process is not adapted for full reverse. We could add a process Prepare_for_Full_topup, which takes the forward and reverse dwis and extracts their b0 separately. An alternative is to add a Extract_rev_b0 process and make its output mix with rev_b0_for_preapre_topup